rtloader/common/stringutils.c

```diff
@@ -65,14 +73,131 @@ char *as_string(PyObject *object)
         return NULL;
     }
 
-    retval = strdupe(PyBytes_AS_STRING(temp_bytes));
-    Py_XDECREF(temp_bytes);
+    retval = PyBytes_AS_STRING(temp_bytes);
+    *stringObject = temp_bytes;
 #endif
 
     return retval;
 }
 
-int init_stringutils(void) {
+/**
+ * returns a C (NULL terminated UTF-8) string from a python string.
+ *
+ * \param object  A Python string to be converted to C-string.
+ *
+ * \return A standard C string (NULL terminated character pointer)
+ *  The returned pointer is allocated from the heap and must be
+ * deallocated (free-ed) by the caller
+ */
+char *as_string(PyObject *object)
+{
+    PyObject *temp_bytes = NULL;
+    char *retval = NULL;
+    char *tmp = NULL;
+
+    if (object == NULL) {
+        return NULL;
+    }
+
+    tmp = as_embedded_string(object, &temp_bytes);
+    if (tmp != NULL && temp_bytes != NULL) {
+        retval = strdupe(tmp);
+        Py_XDECREF(temp_bytes);
+    }
+
+    return retval;
+}
+
+long attr_as_long(PyObject *object, const char *attributeName)
+{
+    if (object == NULL) {
+        return -1;
+    }
+
+    long value = -1;
+    PyObject *py_attr = NULL;
+
+    py_attr = PyObject_GetAttrString(object, attributeName);
+    if (py_attr != NULL) {
+        value = PyLong_AsLong(py_attr);
+    }
+
+    Py_XDECREF(py_attr);
+
+    return value;
+}
+
+size_t attr_as_string_size(PyObject *object, const char *attributeName)
+{
+    PyObject *temp_bytes = NULL;
+    char *tmp = NULL;
+    size_t size = 0;
+
+    if (object == NULL) {
+        return 0;
+    }
+
+    PyObject *py_attr = PyObject_GetAttrString(object, attributeName);
+    if (py_attr != NULL && PyUnicode_Check(py_attr)) {
```

Comment:
Would it make sense to accept PyString too in Py2?

Reply:
I actually do not understand nuances of the question. Can you please elaborate?